backend/query_conversion.go

```diff
@@ -0,0 +1,32 @@
+package backend
+
+import (
+	"context"
+)
+
+// QueryConversionHandler is an EXPERIMENTAL service that allows converting queries between versions
+
+type QueryConversionHandler interface {
+	// ConvertQuery is called to covert queries between different versions
+	ConvertQuery(context.Context, *QueryConversionRequest) (*QueryConversionResponse, error)
+}
+
+type ConvertQueryFunc func(context.Context, *QueryConversionRequest) (*QueryConversionResponse, error)
+
+// ConvertQuery calls fn(ctx, req).
+func (fn ConvertQueryFunc) ConvertQuery(ctx context.Context, req *QueryConversionRequest) (*QueryConversionResponse, error) {
+	return fn(ctx, req)
+}
+
+// QueryConversionRequest supports converting a query from on version to another
+type QueryConversionRequest struct {
+	// NOTE: this may not include app or datasource instance settings depending on the request
```

Comment:
Why not? Or you mean one of them is guaranteed to be populated?

Reply:
this was a copy-paste leftover